### ISSUE TITLE
Issue regarding pickling a convolutional neural network

### ISSUE THREAD

**Issue by ramyabandaru** [lisa-lab/DeepLearningTutorials#151]
I have modified "convolutional_mlp.py" file so that I can pickle the cnn class's object which will further  help me to predict the classes of image using the saved model.But I am getting an error while doing so in the line-377 where I am actually calling the pickle.dump() method.
I am attaching the code for the file and the screenshot of the error.I guess the error is due to pickling the instance methods of logistic_sgd(though I am not sure about it)

self.layer3 = LogisticRegression(input=self.layer2.output, n_in=500, n_out=10)
        self.negative_log_likelihood = (
                self.layer3.negative_log_likelihood
            )
        #predicting the calss value and calculating errors
        self.predict=self.layer3.y_pred
        self.errors = self.layer3.errors
        # the cost we minimize during training is the NLL of the model
        #self.cost = self.layer3.negative_log_likelihood(y)
        # create a list of all model parameters to be fit by gradient descent
        self.params = self.layer3.params + self.layer2.params + self.layer1.params + self.layer0.params
        self.input = input

[covolutional.txt](https://github.com/lisa-lab/DeepLearningTutorials/files/192184/covolutional.txt)
![capture](https://cloud.githubusercontent.com/assets/8428862/14084693/9ccc5256-f539-11e5-84f0-347787a4e038.PNG)

**Comment by nouiz** [comment]
Can you confirm that you use Theano 0.8? If not, update to it.

**Comment by ramyabandaru** [comment]
I have updated the version to 0.8 but still prompts the same error(The CNN I was trying to build above is for CIFAR-10 dataset)

**Comment by ramyabandaru** [comment]
Hi @nouiz 
I have made certain changes in my code and it is  executing without any error but while I am trying to load the pickled model to predict the class labels
[covolutional.txt](https://github.com/lisa-lab/DeepLearningTutorials/files/196241/covolutional.txt)
[predict.txt](https://github.com/lisa-lab/DeepLearningTutorials/files/196242/predict.txt)
![unpickle](https://cloud.githubusercontent.com/assets/8428862/14154191/d5a91e38-f6d8-11e5-949e-b417cce2a1dc.PNG)


 I am getting some error. I am attaching the screenshot of error and the modified file.
Thanks.

**Comment by nouiz** [comment]
In the DLT we hardcode that batch size to make the code easier to understand. But this ask you to respect it. Use the same batch size or update the original code to not hard code the batch size.

**Comment by ramyabandaru** [comment]
I am sorry but I didnot understand what you meant @nouiz  ! what is the batchsize that i have to use now?

**Comment by nouiz** [comment]
If you don't understand what I mean by batch size, I would recommand that
you read all the tutorials before the convolution one. To know the size of
the batch size to use, I would suggest you to read the code. It is written
explicitly. If you want to understand well what is being done, reading and
understanding the code is useful. See this as a small exercice :)

**Comment by ramyabandaru** [comment]
Thanks for your suggestion :) but i do understand what a batchsize is . But I was unable to figureout the problem with it.
